server/src-lib/Hasura/App.hs

```diff
@@ -181,71 +198,64 @@ data Loggers
 newtype AppM a = AppM { unAppM :: IO a }
   deriving (Functor, Applicative, Monad, MonadIO, MonadBase IO, MonadBaseControl IO, MonadCatch, MonadThrow, MonadMask)
 
--- | this function initializes the catalog and returns an @InitCtx@, based on the command given
--- - for serve command it creates a proper PG connection pool
--- - for other commands, it creates a minimal pool
--- this exists as a separate function because the context (logger, http manager, pg pool) can be
--- used by other functions as well
+-- | Initializes the catalog and returns an @InitCtx@ for given @'ServeOptions'
```

Comment:
As a rule of thumb: a comment that does little more than spell out the Haskell type of the thing being documented is not documentation. If the comments contain information that can be extracted from the Haskell code itself, they're not adding information, just duplicating it.

[Here](https://carlanderson.xyz/how-to-write-useful-comments/)'s one of many guides you might find useful to read about writing comments.